### PR TITLE
Add support for Cassandra 3.x

### PR DESCRIPTION
**Pull request by rahulsom**
https://issues.apache.org/jira/browse/CASSANDRA-6717 mentions the changes introduced in Cassandra 3.x.
The connection determines the version of cassandra and runs an appropriate query based on that.

`publishing.gradle` has been tweaked to publish snapshots to local maven without needing `smartThingsArtifactoryUserName`

### CODE REVIEW

**Comment by rahulsom** [comment]
It looks like your `circleci.yml` is set to use java 7. Is java 7 support crucial? Can I bump it up to java 8?

**Comment by beckje01** [comment]
Yes for us at this time we have to have java7 support

File: build.gradle [hunk 1]
```diff
@@ -29,13 +29,13 @@ repositories {
...
 dependencies {
-	compile 'com.datastax.cassandra:cassandra-driver-core:2.1.7.1'
+	compile 'com.datastax.cassandra:cassandra-driver-core:3.0.1'
 	compile 'com.google.guava:guava:18.0'
 
 	testCompile 'org.spockframework:spock-core:1.0-groovy-2.4'
 	testCompile 'org.codehaus.groovy:groovy-all:2.4.4'
 	testRuntime 'io.netty:netty-all:4.1.0.Beta5'
-	testCompile 'org.cassandraunit:cassandra-unit:2.1.9.2'
+	testCompile 'org.cassandraunit:cassandra-unit:3.0.0.1'
```
Comment (beckje01) [inline build.gradle]: If you leave the cassandra unit to 2.1 we should be fine to upgrade the driver as needed.

Comment (rahulsom) [inline build.gradle]: `2.1.9.2` causes this exception:

```
java.lang.NoSuchMethodError: com.datastax.driver.core.Cluster$Builder.addContactPoints(Ljava/lang/String;)Lcom/datastax/driver/core/Cluster$Builder;
	at org.cassandraunit.CassandraCQLUnit.load(CassandraCQLUnit.java:42)
	at org.cassandraunit.BaseCassandraUnit.before(BaseCassandraUnit.java:32)
	at smartthings.migration.MigrationRunnerSpec.setup(MigrationRunnerSpec.groovy:21)
```

So there might not be a way to support cassandra 3 with java 7.

Comment (rahulsom) [inline build.gradle]: In fact `2.2.2.1` has the same problem too.

Comment (rahulsom) [inline build.gradle]: My gradle-fu is not too strong, but the whole problem happens to be in the test dependencies. 
However, I know that we can set the `sourceVersion` and `targetVersion` on the build. So I think we could create jars that are java7 compatible, but use java8 for the build process.

Is that something you would be comfortable with?

**Comment by beckje01** [comment]
I'll have some time to dig into this in the afternoon. I'll see what I can do.

**Comment by jterhune** [comment]
I was able to override the version of cassandra-all to 2.x on my fork, but it failed on this cassandra-unit issue: https://github.com/jsevellec/cassandra-unit/pull/177

The good news is that PR is already merged to cassandra-unit master... the bad news is that it's not released.

My fork for reference: https://github.com/jterhune/cassandra-migration

**Comment by beckje01** [comment]
I am just going to roll this project forward to java 8

**Pull request merged**